python_modules/dagster/dagster/components/core/load_defs.py

```diff
@@ -117,6 +117,22 @@ def defs():
                 project_path = Path("/path/to/my/dg/project")
                 return dg.load_from_defs_folder(project_root=project_path)
 
+    """
+    return component_tree_for_project(project_root=project_root).build_defs()
+
+
+@public
+@suppress_dagster_warnings
+def component_tree_for_project(*, project_root: Path) -> ComponentTree:
```

Comment:
did you see `ComponentTree.load` ? 
https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/components/core/tree.py#L140-L144

slightly more flexible than this, maybe could use a better name

Reply:
actually, realizing this is a bit more finicky bc it requires `DgContext`, so I don't think it's a drop in replacement

Reply:
it looks like the bits we use `DgContext` for are what is currently in the body of this function, maybe move this code in to replace that?

Reply:
yeah, that makes sense - updated, & renamed to `for_project` since I think that's a little more explicit & avoids the `load` language which we use for components themselves